platform/src/main/java/org/stellar/anchor/platform/config/ClientsConfig.java

```diff
@@ -0,0 +1,85 @@
+package org.stellar.anchor.platform.config;
+
+import static org.stellar.anchor.util.StringHelper.isEmpty;
+
+import lombok.Data;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+import org.stellar.anchor.api.exception.SepException;
+import org.stellar.anchor.sep10.Sep10Helper;
+
+@Data
+public class ClientsConfig implements Validator {
+  ClientType type;
+  String signingKey;
+  String domain;
+  String callbackUrl;
+
+  public enum ClientType {
+    CUSTODIAL,
+    NONCUSTODIAL
+  }
+
+  @Override
+  public boolean supports(@NotNull Class<?> clazz) {
+    return ClientsConfig.class.isAssignableFrom(clazz);
+  }
+
+  @Override
+  public void validate(@NotNull Object target, @NotNull Errors errors) {
+    ClientsConfig config = (ClientsConfig) target;
+
+    if (config.type.equals(ClientType.CUSTODIAL)) {
+      validateCustodialClient(errors);
+    } else {
+      validateNonCustodialClient(errors);
+    }
+  }
+
+  void validateCustodialClient(Errors errors) {
+    ValidationUtils.rejectIfEmptyOrWhitespace(
+        errors,
+        "signingKey",
+        "empty-client-signing-key",
+        "The client.signingKey cannot be empty and must be defined");
+
+    ValidationUtils.rejectIfEmptyOrWhitespace(
+        errors,
+        "callbackUrl",
+        "empty-client-callback-url",
+        "The client.callbackUrl cannot be empty and must be defined");
+  }
```

Comment:
Callback URLs should be optional. Some wallets may opt to poll instead, or may use polling first before implementing callbacks at a later stage. Same goes for noncustodial wallets.

Reply:
In that case, please also update the comments of the config YAML file.